src-tauri/src/main.rs

```diff
@@ -33,9 +33,16 @@ fn generate_primers(
 }
 
 #[tauri::command]
-fn encode_sequence(encoder_type: &str, file_path: &str) -> Result<Vec<Base>, String> {
-    let bytes = fs::read(file_path).map_err(|err| err.to_string())?;
-    let bits = BitVec::<_, Msb0>::from_vec(bytes);
+fn encode_sequence(
+    encoder_type: &str,
+    file_path: PathBuf,
+) -> Result<Vec<Base>, String> {
+    let compressor = VoidCompressor{}; 
+    let compressed = compressor
+        .compress(file_path)
+        .map_err(|err| err.to_string())?;
+    let bytes = fs::read(compressed).map_err(|err| err.to_string())?;
+    let bits = BitVec::<_, Msb0>::from_slice(&bytes);
```

Comment:
To test this, do the below
![image](https://github.com/user-attachments/assets/6f9bf237-7adb-4b40-b771-89d5039a292b)